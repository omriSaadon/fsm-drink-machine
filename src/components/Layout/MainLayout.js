import React from 'react';
import PropTypes from 'prop-types';
import classes from './MainLayout.scss';

const propTypes = {
  children: PropTypes.node,
};

const MainLayout = ({ children }) => (
  <div className={classes.wrapper}>
    <div className={classes.content}>
      {children}
    </div>
  </div>
);

MainLayout.propTypes = propTypes;
export default MainLayout;
