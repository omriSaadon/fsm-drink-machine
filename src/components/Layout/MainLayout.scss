
$header-height: 50px;
$white: #ffffff;
$contrast: #f6f6f6;

body {
  margin: 0;
}

* {
  box-sizing: border-box;
}

.wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: $contrast;
}

.content {
  height: 100%;
  width: 1170px;
  box-shadow: 0 2px 4px 0 rgba(17, 17, 17, .2);
  background-color: $white;
  padding: 10px 50px 55px;
}
