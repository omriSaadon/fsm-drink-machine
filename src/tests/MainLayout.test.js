import React from 'react';
import { render, screen } from '@testing-library/react';
import MainLayout from 'components/Layout/MainLayout';

test('Verify text is in layout', () => {
  render(<MainLayout>Test text in layout</MainLayout>);
  const linkElement = screen.getByText(/Test text in layout/i);
  expect(linkElement).toBeInTheDocument();
});

test('Verify text is NOT in layout', () => {
  render(<MainLayout>Test text in layout</MainLayout>);
  const linkElement = screen.queryByText(/some other text/i);
  expect(linkElement).not.toBeInTheDocument();
});

test('Verify snapshot', () => {
  const { asFragment } = render(<MainLayout>Test text in layout</MainLayout>);
  expect(asFragment()).toMatchSnapshot();
});
