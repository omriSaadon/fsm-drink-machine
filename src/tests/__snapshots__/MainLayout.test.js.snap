// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Test text in layout
    </div>
  </div>
</DocumentFragment>
`;
